label-application-rule-config-service-impl/src/main/java/org/hypertrace/label/application/rule/config/service/LabelApplicationRuleConfigServiceImpl.java

```diff
@@ -27,29 +37,42 @@
 
 public class LabelApplicationRuleConfigServiceImpl
     extends LabelApplicationRuleConfigServiceGrpc.LabelApplicationRuleConfigServiceImplBase {
-  static final String LABEL_APPLICATION_RULE_CONFIG_SERVICE_CONFIG =
+  private static final String LABEL_APPLICATION_RULE_CONFIG_SERVICE_CONFIG =
       "label.application.rule.config.service";
-  static final String MAX_DYNAMIC_LABEL_APPLICATION_RULES_PER_TENANT =
+  private static final String MAX_DYNAMIC_LABEL_APPLICATION_RULES_PER_TENANT =
       "max.dynamic.label.application.rules.per.tenant";
-  static final int DEFAULT_MAX_DYNAMIC_LABEL_APPLICATION_RULES_PER_TENANT = 100;
+  private static final String SYSTEM_LABEL_APPLICATION_RULES = "system.label.application.rules";
+  private static final int DEFAULT_MAX_DYNAMIC_LABEL_APPLICATION_RULES_PER_TENANT = 100;
+
   private final IdentifiedObjectStore<LabelApplicationRule> labelApplicationRuleStore;
   private final LabelApplicationRuleValidator requestValidator;
   private final int maxDynamicLabelApplicationRulesAllowed;
+  private final List<LabelApplicationRule> systemLabelApplicationRules;
+  private final Map<String, LabelApplicationRule> systemLabelApplicationRulesMap;
 
   public LabelApplicationRuleConfigServiceImpl(
       Channel configChannel, Config config, ConfigChangeEventGenerator configChangeEventGenerator) {
-    int maxDynamicRules = DEFAULT_MAX_DYNAMIC_LABEL_APPLICATION_RULES_PER_TENANT;
-    if (config.hasPath(LABEL_APPLICATION_RULE_CONFIG_SERVICE_CONFIG)) {
-      Config labelApplicationRuleConfig =
-          config.getConfig(LABEL_APPLICATION_RULE_CONFIG_SERVICE_CONFIG);
-      if (labelApplicationRuleConfig.hasPath(MAX_DYNAMIC_LABEL_APPLICATION_RULES_PER_TENANT)) {
-        maxDynamicRules =
-            labelApplicationRuleConfig.getInt(MAX_DYNAMIC_LABEL_APPLICATION_RULES_PER_TENANT);
-      }
+    Config labelApplicationRuleConfig =
```

Comment:
Can we please move all this new code out of here. We don't want to overload business logic into the grpc impls, they're just routers. Please create a new config class that handles loading these values from config.

Reply:
Done